app/models/package_manager/maven.rb

```diff
@@ -109,45 +110,58 @@ def self.mapping(raw_project, depth = 0)
       nil
     end
 
-    def self.mapping_from_pom_xml(version_xml, depth = 0)
-      xml = if version_xml.respond_to?("project")
-              version_xml.project
-            else
-              version_xml
-            end
-
-      parent = {
-        description: nil,
-        homepage: nil,
-        repository_url: "",
-        licenses: "",
-        properties: {},
-      }
-      if xml.locate("parent").present? && depth < MAX_DEPTH
-        group_id = extract_pom_value(xml, "parent/groupId")&.strip
-        artifact_id = extract_pom_value(xml, "parent/artifactId")&.strip
-        version = extract_pom_value(xml, "parent/version")&.strip
-        if group_id && artifact_id && version
-          parent = mapping_from_pom_xml(
-            get_pom(group_id, artifact_id, version),
-            depth + 1
-          )
+    def self.get_pom_documents(group_id, artifact_id, version, depth = 0)
+      return [] if depth >= MAX_DEPTH
```

Comment:
as long as we're backfilling we might consider increasing `MAX_DEPTH` ... it really just needs to catch hostile stuff, but we want to support any real-world thing right? Why not like 50 instead of 5?

Reply:
My example POM was actually getting pretty close to this limit. Bumped to 50. I also confirmed that the MAX_DEPTH in Bibliothecary doesn't look related to parent pom depth, and it's ok that they're different.

Reply:
that's a great point -- theoretically we could run through all Maven packages and estimate their depth, to find good boundaries